### DISCUSSION TITLE
How does milvus pick the chunks from its database

### DISCUSSION THREAD

**Discussion by jamesjohnson1025** [milvus-io/milvus#25665] [Q&A and General discussion] [answered]
HI, i wish to know how milvus pick the chunks. I have a pdf file and i split the pdf into multiple chunks and stored them in milvus.
The pdf contains questions and answers. There was a question `Can i get placed in current location` in pdf. I queried milvus the exact same question. The expectation was at least the chunk with  that question get's picked.Unfortunately No.  None of the top 5 chunks had the sentence `Can i get placed in current location` .

NOTE: I had computed the embeddings of `Can i get placed in current location` and queried milvus db for results.

Below are the search params

```
 self.query_params = {
            'anns_field': 'embeddings',
            'param': {
                'metric_type': 'IP',
                'params': {
                    'nprobe': 50
                }
            },
            'offset': 0,
            'limit': 2,
            'output_fields': ['*'],
            'consistency_level': 'Strong'
        }
```
Am i missing something here ? Can you help me ?. Thank you. Do i have to tweak any parameter ?. If so, please let me know.

**Comment by yhmo** [comment, marked as answer]
To use the 'metric_type': 'IP', you need to make sure all the embeddings are normalized(each embedding length is 1.0).
If your embeddings are not normalized, then you should use 'metric_type': 'L2' to calculate distance.

FYI: https://milvus.io/docs/metric.md#Similarity-Metrics
